graphql_compiler/tests/test_explain_info.py

```diff
@@ -0,0 +1,95 @@
+        expected_filters = dict(expected_filters)
+        expected_recurses = dict(expected_recurses)
+        for location, _ in meta.registered_locations:
+            # Do filters match with expected for this location?
+            filters = meta.get_filter_infos(location)
+            self.assertEqual(filters, expected_filters.get(location, []))
+            if filters:
+                del expected_filters[location]
+            # Do recurse match with expected for this location?
+            recurse = meta.get_recurse_infos(location)
+            self.assertEqual(recurse, expected_recurses.get(location, []))
+            if recurse:
+                del expected_recurses[location]
+        # Any expected infos missing?
+        self.assertEqual(0, len(expected_filters))
+        self.assertEqual(0, len(expected_recurses))
+
+    def test_filter(self):
+        self.check(test_input_data.traverse_filter_and_output,
+                   [
+                       (Location(('Animal', 'out_Animal_ParentOf'), None, 1),
+                        [FilterInfo(field_name='out_Animal_ParentOf',
+                                    op_name='name_or_alias',
+                                    args=['$wanted'])]),
+                   ],
+                   [
+                   ])
+
+    def test_filters(self):
+        self.check(test_input_data.complex_optional_traversal_variables,
+                   [
+                       (Location(('Animal',), None, 1),
+                        [FilterInfo(field_name='name',
+                                    op_name='=',
+                                    args=['$animal_name'])]),
+                       (Location(('Animal', 'in_Animal_ParentOf', 'out_Animal_FedAt'), None, 1),
+                        [FilterInfo(field_name='name',
+                                    op_name='=',
+                                    args=['%parent_fed_at_event']),
+                         FilterInfo(field_name='event_date',
+                                    op_name='between',
+                                    args=['%other_child_fed_at', '%parent_fed_at'])]),
+                   ],
+                   [
+                   ])
+
+    def test_fold(self):
+        self.check(test_input_data.coercion_filters_and_multiple_outputs_within_fold_scope,
+                   [],
+                   [])
+
+    def test_recurse(self):
+        self.check(test_input_data.simple_recurse,
+                   [
+                   ],
+                   [
+                       (Location(('Animal',), None, 1),
+                        [RecurseInfo(edge_direction='out', edge_name='Animal_ParentOf', depth=1)]),
+                   ])
+
+    def test_two_recurses(self):
+        self.check(test_input_data.two_consecutive_recurses,
+                   [
+                       (Location(('Animal',), None, 1),
+                        [FilterInfo(field_name='Animal',
```

Comment:
I'd be particularly interested in a test case that includes the `has_edge_degree` filter -- both what location it appears as assigned to, and what `field_name` carries in that case. We might have to tweak how data is passed around / what data `FilterInfo` carries based on that.

This is my only major concern with this PR at the moment.